include/boost/histogram/python/register_histogram.hpp

```diff
@@ -171,16 +171,25 @@ register_histogram(py::module& m, const char* name, const char* desc) {
 
         .def(
             "axis",
-            [](const histogram_t& self, int i) {
+            [](const histogram_t& self, int i) -> py::object {
                 unsigned ii = i < 0 ? self.rank() - (unsigned)std::abs(i) : (unsigned)i;
-                if(ii < self.rank())
-                    return self.axis(ii);
+
+                if(ii < self.rank()) {
+                    const axis_variant& var = bh::unsafe_access::axis(self, ii);
```

Comment:
this is basically what i had in mind as well. only a minor comment: why use unsafe_access here instead of `self.axis(ii)`?

Reply:
apart from that this looks very good

Reply:
Leftover from when I first tried to avoid const, since the py::cast is going to ignore the const. Fixed.